--- ./ext/network_interface_ext/netifaces.c
+++ ./ext/network_interface_ext/netifaces.c
@@ -677,7 +677,7 @@
     return Qnil;
   }
 
-  *pfreq = ifc.CNAME(ifc_req);
+  pfreq = ifc.CNAME(ifc_req);
 
   for (n = 0; n < ifc.CNAME(ifc_len)/sizeof(struct CNAME(ifreq));n++,pfreq++)
   {
--- ./ext/network_interface_ext/netifaces.h
+++ ./ext/network_interface_ext/netifaces.h
@@ -30,6 +30,7 @@
 /* For Linux, include all the sockaddr
    definitions we can lay our hands on. */
 #if !HAVE_SOCKADDR_SA_LEN
+#    include <netinet/in.h>
 #  if HAVE_NETASH_ASH_H
 #    include <netash/ash.h>
 #  endif
