diff -uNr Python-3.6.2/Lib/webbrowser.py Python-3.6.2.mod/Lib/webbrowser.py
--- Python-3.6.2/Lib/webbrowser.py	2017-07-08 06:33:27.000000000 +0300
+++ Python-3.6.2.mod/Lib/webbrowser.py	2017-09-05 21:05:56.823946902 +0300
@@ -1,4 +1,4 @@
-#! /usr/bin/env python3
+#!/data/data/com.termux/files/usr/bin/env python3
 """Interfaces for launching and remotely controlling Web browsers."""
 # Maintained by Georg Brandl.
 
@@ -501,6 +501,9 @@
     # The w3m browser <http://w3m.sourceforge.net/>
     if shutil.which("w3m"):
         register("w3m", None, GenericBrowser("w3m"))
+    # If using termux, try to open Android's browser
+    if shutil.which("termux-open-url"):
+        register("termux-open-url", None, GenericBrowser("termux-open-url"))
 
 #
 # Platform support for Windows
