diff -uNr metasploit-credential-2.0.12/spec/lib/metasploit/credential/importer/zip_spec.rb metasploit-credential-2.0.12.mod/spec/lib/metasploit/credential/importer/zip_spec.rb
--- metasploit-credential-2.0.12/spec/lib/metasploit/credential/importer/zip_spec.rb	2017-09-11 16:39:57.475959843 +0300
+++ metasploit-credential-2.0.12.mod/spec/lib/metasploit/credential/importer/zip_spec.rb	2017-09-11 16:42:26.075959736 +0300
@@ -6,7 +6,7 @@
   subject(:zip_importer){ FactoryGirl.build :metasploit_credential_importer_zip, workspace: workspace }
 
   describe "validations" do
-    DUMMY_ZIP_PATH = "/tmp/import-test-dummy.zip"
+    DUMMY_ZIP_PATH = "/data/data/com.termux/files/usr/tmp/import-test-dummy.zip"
 
     context "when the zip file contains a keys directory and a manifest CSV" do
       it { is_expected.to be_valid }
@@ -64,4 +64,4 @@
       expect(Metasploit::Credential::Importer::Zip::ZIP_HEADER_IDENTIFIER.size).to eq(Metasploit::Credential::Importer::Zip::ZIP_HEADER_BYTE_LENGTH)
     end
   end
-end
\ No newline at end of file
+end
diff -uNr metasploit-credential-2.0.12/spec/support/shared/contexts/metasploit/credential/importer/zip/file.rb metasploit-credential-2.0.12.mod/spec/support/shared/contexts/metasploit/credential/importer/zip/file.rb
--- metasploit-credential-2.0.12/spec/support/shared/contexts/metasploit/credential/importer/zip/file.rb	2017-09-11 16:39:57.485959843 +0300
+++ metasploit-credential-2.0.12.mod/spec/support/shared/contexts/metasploit/credential/importer/zip/file.rb	2017-09-11 16:42:31.755959732 +0300
@@ -5,7 +5,7 @@
 
   def remove_metasploit_credential_importer_zip_files
     # Using POSIX filepath here b/c we don't care about RSpec on Windows
-    Dir.glob("/tmp/metasploit_credential_importer_zip_file*") do |path|
+    Dir.glob("/data/data/com.termux/files/usr/tmp/metasploit_credential_importer_zip_file*") do |path|
       FileUtils.rm_rf(path)
     end
   end
@@ -23,4 +23,4 @@
   after(:example) do
     remove_metasploit_credential_importer_zip_files
   end
-end
\ No newline at end of file
+end
